src/polymer/behavior-scanner.ts

```diff
@@ -105,15 +110,22 @@ class BehaviorVisitor implements Visitor {
       if (name in this.propertyHandlers) {
         this.propertyHandlers[name](prop.value);
       } else if (babel.isMethod(prop) || babel.isFunction(prop.value)) {
-        const method = esutil.toScannedMethod(
-            prop, this.document.sourceRangeForNode(prop)!, this.document);
-        this.currentBehavior.addMethod(method);
+        if (!babel.isMethod(prop) || (prop.kind !== 'get' && prop.kind !== 'set')) {
```

Comment:
The `!babel.isMethod(prop)` part here doesn't seem right. We only want to filter out accessors here, don't we?

Reply:
`!babel.isMethod` is because it could be a `Function`, and the second condition is for if it is a method, we want to check `Method#kind` isn't an accessor.

Reply:
I'm saying we should either call `addMethod` for functions (like we are currently), or we should drop the `|| babel.isFunction` part on line 112.

I think functions on the behavior object should be treated as methods, even if they don't use method syntax.

Reply:
👀

Reply:
👋 
so from what you're saying, i _think_ you mean i should put another `else if (babel.isFunction(prop.value))` and treat them as methods?

this double-if makes me twitch too. the logic looks right but its ugly. the `!babel.isMethod` is because, if its not a method, it must be a function... and if it is a method, it has a `kind` so we check if its an accessor

Reply:
its like...

```
if (method or function) {
  if (function or method_which_isnt_an_accessor) {
    do_a_method;
  } else {
    do_a_property;
  }
}
```